### DISCUSSION TITLE
How to get rendered conifg template in custom script

### DISCUSSION THREAD

**Discussion by PaulR282** [netbox-community/netbox#14562] [Q&A] [answered]
How can I get the rendered config template of a device in a custom script?

**Comment by candlerb** [comment, marked as answer]
[class Device](https://github.com/netbox-community/netbox/blob/v3.6.7/netbox/dcim/models/devices.py#L529-L536) inherits from from [RenderConfigMixin](https://github.com/netbox-community/netbox/blob/v3.6.7/netbox/dcim/models/mixins.py#L54-L75)

```
# /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell
### NetBox interactive shell (netbox3)
### Python 3.8.10 | Django 4.2.8 | NetBox 3.6.7
### lsmodels() will show available models. Use help(<model>) for more info.
>>> help(Device)
...
|  ----------------------------------------------------------------------
 |  Methods inherited from dcim.models.mixins.RenderConfigMixin:
 |
 |  get_config_template(self)
 |      Return the appropriate ConfigTemplate (if any) for this Device.
 |
 |  ----------------------------------------------------------------------
...
>>> d = Device.objects.get(name="gw2")
>>> d.get_config_template()
<ConfigTemplate: Mikrotik interface vlans>
>>>
```

And ConfigTemplate has a `render` method, to which you pass the local variables available within that template:

```
>>> d.get_config_template().render({"device":d})
'\n/interface bridge vlan add bridge=bridge ...
```

**Reply by PaulR282** [reply]
Do you also know how I can render with config context? With just `render({"device":d})` it doesn't render it with the conext for me. If I do `render({"device":device}, context=device.get_config_context())` it errores saying `TypeError: ConfigTemplate.render() got multiple values for argument 'context'`

**Reply by candlerb** [reply]
I guess you [copy what Netbox does](https://github.com/netbox-community/netbox/blob/v3.7.0/netbox/dcim/views.py#L2046-L2082), which means you merge them, something like this:

```
>>> d = Device.objects.get(name="gw2")
>>> context_data = d.get_config_context()
>>> context_data.update({"device": d})
>>> d.get_config_template().render(context=context_data)
```

**Reply by PaulR282** [reply]
That worked. Thanks.
